### PR TITLE
feat: op-contracts `L2OutputOracle.sol` add EOTS info 

### PR DESCRIPTION
**Pull request by lesterli**
## Summary

* define `EOTSInfo` in `Types.sol` and update `L2OutputOracle.sol` to receive the data from `op-proposer`
* `npm run snapshots` also generates bytecode of contract

## Test Plan

run `npm run snapshots` under the `packages/contracts-bedrock`

### CODE REVIEW

File: packages/contracts-bedrock/scripts/autogen/generate-snapshots.ts [hunk 1]
```diff
@@ -77,14 +77,23 @@ const parseArtifactName = (artifactVersionFile: string): string => {
...
 
   const storageLayoutDir = path.join(outdir, 'storageLayout')
   const abiDir = path.join(outdir, 'abi')
-  fs.rmSync(storageLayoutDir, { recursive: true })
-  fs.rmSync(abiDir, { recursive: true })
+  const binDir = path.join(outdir, 'bin')
+  if (fs.existsSync(storageLayoutDir)) {
+    fs.rmSync(storageLayoutDir, { recursive: true })
+  }
+  if (fs.existsSync(abiDir)) {
```
Comment (bap2pecs) [inline packages/contracts-bedrock/scripts/autogen/generate-snapshots.ts]: why adding `existsSync`?

File: packages/contracts-bedrock/src/L1/L2OutputOracle.sol [hunk 2]
```diff
@@ -231,6 +232,10 @@ contract L2OutputOracle is Initializable, ISemver {
             );
         }
 
+        for (uint i = 0; i < _eotsInfos.length; i++) {
+            require(_eotsInfos[i].fpBtcPk != bytes32(0), "L2OutputOracle: EOTSInfo.fpBtcPk cannot be zero");
```
Comment (bap2pecs) [inline packages/contracts-bedrock/src/L1/L2OutputOracle.sol:236]: let's add a TODO comment so we know we will add a call here

File: packages/contracts-bedrock/test/invariants/L2OutputOracle.t.sol [hunk 3]
```diff
@@ -25,7 +26,16 @@ contract L2OutputOracle_Proposer {
...
+
+        Types.EOTSInfo[] memory _eotsInfos = new Types.EOTSInfo[](1);
+        Types.EOTSInfo memory eotsInfo = Types.EOTSInfo({
+            fpBtcPk: keccak256(abi.encode("fpBtcPk")),
+            pubRand: keccak256(abi.encode("pubRand")),
+            pubRandProof: new bytes32[](1),
+            finalitySig: keccak256(abi.encode("finalitySig"))
+        });
+        _eotsInfos[0] = eotsInfo;
+        oracle.proposeL2Output(_outputRoot, _l2BlockNumber, _l1BlockHash, _l1BlockNumber, _eotsInfos);
```
Comment (bap2pecs) [inline packages/contracts-bedrock/test/invariants/L2OutputOracle.t.sol:38]: why not just use `_defaultEotsInfos` here?

File: packages/contracts-bedrock/src/L1/L2OutputOracle.sol [hunk 4]
```diff
@@ -197,7 +197,8 @@ contract L2OutputOracle is Initializable, ISemver {
         bytes32 _outputRoot,
         uint256 _l2BlockNumber,
         bytes32 _l1BlockHash,
-        uint256 _l1BlockNumber
+        uint256 _l1BlockNumber,
+        Types.EOTSInfo[] calldata _eotsInfos
```
Comment (SebastianElvis) [inline packages/contracts-bedrock/src/L1/L2OutputOracle.sol:201]: docstring needs to be updated w.r.t. `_eotsInfos` as well

**Review by SebastianElvis — APPROVED** [review]

**Review by bap2pecs — CHANGES_REQUESTED** [review]
won't touch l2oo per https://github.com/ethereum-optimism/specs/discussions/218

**Pull request closed without merging**